### ISSUE TITLE
The Shader is broken on android

### ISSUE THREAD

**Issue by MrAgentBlaze** [depressed-pho/natural-mystic-shaders#62]
The Shadet works in my new Samsung Galaxy A5 android 6.0.1 . You can see the pic i sent to pho (you) in discord

**Comment by depressed-pho** [comment]
Yes I saw it. The sky shader is working flawlessly but the terrain is rendered completely black. I have no idea why that happens. Someone please shed some light.

**Comment by gio1135** [comment]
It works perfectly fine for me. I'm using a Pixel 2 XL.

**Comment by gio1135** [comment]
> Yes I saw it. The sky shader is working flawlessly but the terrain is rendered completely black. I have no idea why that happens. Someone please shed some light.
> 
> ![](https://camo.githubusercontent.com/5ff61f5979af814c21662bc136f90864f42f951a/68747470733a2f2f63646e2e646973636f72646170702e636f6d2f6174746163686d656e74732f3533353339323332393335323231363538362f3534343133343530353836333138343432342f53637265656e73686f745f323031392d30322d31302d31392d32392d34392e706e67)

I see what you did ther. Here, let me do so `/effect @s night_vision 10000 255 true`.

**Comment by depressed-pho** [comment]
In that screenshot the far terrain (where far chunk fogs normally appear) has some random colors like yellow, green, and red. This smells like a tone mapping curve being incorrectly applied, possibly due to precision loss of floating point arithmetic. I'll see if it can ever happen in my code, but it's most likely a driver bug.

**Comment by depressed-pho** [comment]
In that screenshot the far terrain (where far chunk fogs normally appear) has some random colors like yellow, green, and red. This smells like a tone mapping curve being incorrectly applied, possibly due to precision loss of floating point arithmetic. I'll see if it can ever happen in my code, but it's most likely a driver bug.
